### ISSUE TITLE
Scoping Activation 

### ISSUE THREAD

**Issue by dfilaretti** [runtimeverification/solidity#61]
Seems we have a problem with variable scoping, as shown by test `C99_scoping_activation` (and possibly others, which will be added here as they are found), 

For example, consider a call to the following function:

```
function g() pure public returns (uint x) {
  x = 7;
  {
      x = 3;
      uint x;
      return x; // This should return the new variable, i.e. 0
  }
}
```

Instead, we return `3`. See `C99_scoping_activation` for more cases,

**Comment by theo25** [comment]
According to documentation, this should return 0 (per C99 scoping rules) when compiled with solidity 0.5 and newer. We are still based on a 0.4 release, so that's why I implemented it to return 3 (per Solidity <0.5 scoping rules). Do they have tests that expect this to return 0 already? If that's the case, we indeed have a bug, as has their documentation.

**Comment by dfilaretti** [comment]
Yes, the code above is copied from the test. 

The check is:
```
ABI_CHECK(callContractFunction("g()"), encodeArgs(0));
```

but it fails because we return 3.
